plugins/aws-appmesh/config/netconfig.go

```diff
@@ -129,16 +129,20 @@ func validateConfig(config netConfigJSON) error {
 	if config.IgnoredGID == "" && config.IgnoredUID == "" {
 		return fmt.Errorf("missing required parameter ignoredGID or ignoredUID")
 	}
-	if len(config.AppPorts) == 0 {
-		return fmt.Errorf("missing required parameter appPorts")
-	}
 	if config.ProxyEgressPort == "" {
 		return fmt.Errorf("missing required parameter proxyEgressPort")
 	}
-	if config.ProxyIngressPort == "" {
+
+	// AppPorts and ProxyIngressPort go in pairs,
+	// i.e. either both are set or both are not set.
+	if config.ProxyIngressPort == "" && len(config.AppPorts) > 0 {
```

Comment:
nit: the error message is not clear enough for customers to know the relationship between appPorts and proxyIngressPort - either both set or not set.

Reply:
Updated to hopefully be a little more clear. :)